nifi-commons/nifi-expression-language/src/main/java/org/apache/nifi/attribute/expression/language/evaluation/functions/GeohashDecodeBaseEvaluator.java

```diff
@@ -0,0 +1,97 @@
+
+import org.apache.commons.lang3.EnumUtils;
+import org.apache.nifi.attribute.expression.language.EvaluationContext;
+import org.apache.nifi.attribute.expression.language.evaluation.Evaluator;
+import org.apache.nifi.attribute.expression.language.evaluation.NumberEvaluator;
+import org.apache.nifi.attribute.expression.language.evaluation.NumberQueryResult;
+import org.apache.nifi.attribute.expression.language.evaluation.QueryResult;
+import org.apache.nifi.attribute.expression.language.exception.AttributeExpressionLanguageException;
+
+import ch.hsr.geohash.GeoHash;
+import ch.hsr.geohash.WGS84Point;
+
+public abstract class GeohashDecodeBaseEvaluator extends NumberEvaluator {
+
+    public enum GeohashFormat {
+        BASE32, BINARY, LONG
+    }
+
+    public enum GeoCoord {
+        LATITUDE, LONGITUDE
+    }
+
+    private final Evaluator<String> subject;
+    private final Evaluator<String> format;
+
+    public GeohashDecodeBaseEvaluator(final Evaluator<String> subject, final Evaluator<String> format) {
+        this.subject = subject;
+        this.format = format;
+    }
+
+    protected QueryResult<Number> geohashDecodeEvaluate(final EvaluationContext evaluationContext, final GeoCoord geoCoord) {
+        //Optional argument. If not specified, defaults to BASE_32_STRING.
+        final GeohashFormat geohashFormatValue;
+        if (format != null) {
+            if(!EnumUtils.isValidEnum(GeohashFormat.class, format.evaluate(evaluationContext).getValue())) {
+                throw new AttributeExpressionLanguageException("Format values must be 'BASE32', 'BINARY' or 'LONG'");
+            }
+            geohashFormatValue = GeohashFormat.valueOf(format.evaluate(evaluationContext).getValue());
+        } else {
+            geohashFormatValue = GeohashFormat.BASE32;
+
+        }
+
+        final Long geohashLongValue = geohashFormatValue == GeohashFormat.LONG ? Long.valueOf(subject.evaluate(evaluationContext).getValue()) : null;
+        final String geohashStringValue = (geohashFormatValue == GeohashFormat.BASE32
+                || geohashFormatValue == GeohashFormat.BINARY) ? subject.evaluate(evaluationContext).getValue() : null;
+
+        if (geohashLongValue == null && geohashStringValue == null) {
+            return new NumberQueryResult(null);
+        }
+
+        try {
+            WGS84Point boundingBoxCenter;
+            switch (geohashFormatValue) {
+                case LONG:
+                    String binaryString = Long.toBinaryString(geohashLongValue);
+                    boundingBoxCenter = GeoHash.fromBinaryString(binaryString).getBoundingBoxCenter();
+                    break;
+                case BINARY:
+                    boundingBoxCenter = GeoHash.fromBinaryString(geohashStringValue).getBoundingBoxCenter();
+                    break;
+                default:
+                    boundingBoxCenter = GeoHash.fromGeohashString(geohashStringValue).getBoundingBoxCenter();
```

Comment:
In testing with invalid data (e.g. "cat") this will throw a null pointer exception. We should do a null check before calling the getBoundingBoxCenter methods.